### ISSUE TITLE
WN34 Soil Temperature Sensor

### ISSUE THREAD

**Issue by Pres499** [garbled1/homeassistant_ecowitt#80]
A new Ecowitt sensor, WN34 Soil Temperature, has been released that is not
included in the current Ecowitt integration.
What are the lines of code and placement that should be added to each specific file 
in this integration for this new WN34 sensor?
This would be helpful in the future for users to update the files for additional new
sensors Ecowitt releases for the GW1000 or GW1100 weather station.

**Comment by lgeitner** [comment]
I am having the same issue I have added an ECOWITT WN34 Soil Temperature Sensor to my GW1000 Gateway.  

The WN34 Soil Temperature and Battery Voltage Show up on https://www.ecowitt.net.  See screen shots below

![WN34SoilTemperature](https://user-images.githubusercontent.com/89822625/133350246-e69eafad-f12d-4551-8bf6-678063a3e88e.JPG)
![WN34BatteryVoltage](https://user-images.githubusercontent.com/89822625/133350247-5eb84838-4877-4f32-8b72-63dd583d026d.JPG)

I get an error in HomeAssistant that it is an unhandled sensor type.

2021-09-14 19:52:18 WARNING (MainThread) [custom_components.ecowitt] Unhandled sensor type tf_ch1
2021-09-14 19:52:18 WARNING (MainThread) [custom_components.ecowitt] Unhandled sensor type tf_batt1
2021-09-14 19:52:49 WARNING (MainThread) [custom_components.ecowitt] Unhandled sensor type tf_ch1 value 71.1, file a PR.
2021-09-14 19:52:49 WARNING (MainThread) [custom_components.ecowitt] Unhandled sensor type tf_batt1 value 1.40, file a PR.

**Comment by Pres499** [comment]
Adding a new sensor from Ecowitt such as the Soil Temp would be a learning experience if we knew the critical files, sensors.py, ecowitt.py or the other integration program files to add the correct commands.  The tf_ch1 value is what is needed to added to the integration.  Maybe, garbled1 can guide us or add this sensor to the integration? 
Your other unhandled sensor, tf_batt1, could be ignored and added to the ignored sensors in the file const.py.

**Comment by lgeitner** [comment]
Here is what I did to get this working.  

This is by no means the best way to solve the issue.

In my case I wanted the battery status of my WN34 Soil Temperature.

This assumes you will never use Temperature 8 or soilbatt1.

Make a backup of file const.py before you proceed!

in the const.py file locate line 9 add a new line that says

    TEMP_FAHRENHEIT,

so the import section will look like this:

    TEMP_CELSIUS,
    TEMP_FAHRENHEIT,

Locate the line TYPE_TEMP8C = "temp8c" and change it to TYPE_TEMP8C = "tf_ch1"
Locate the line TYPE_SOILBATT1 = "soilbatt1" and change it to TYPE_SOILBATT1 = "tf_batt1"

Locate the line     

TYPE_TEMP8C: ("Temperature 8", TEMP_CELSIUS,
                  TYPE_SENSOR, DEVICE_CLASS_TEMPERATURE, "mdi:thermometer", 0),

change it to:

    TYPE_TEMP8C: ("Temperature 8", **TEMP_FAHRENHEIT**,
                  TYPE_SENSOR, DEVICE_CLASS_TEMPERATURE, "mdi:thermometer", 0),

Restart Home Assistant and the soil temperature from the WN34 will show up under sensor.temperature_8 and the battery status of the WN34 Sensor will show up as sensor.soil_moisture_1_battery.

![Capture](https://user-images.githubusercontent.com/89822625/133528378-8f489e8e-639d-45d1-a080-620c362e53ce.JPG)

**Comment by lgeitner** [comment]
It should read as this:

TYPE_TEMP8C: ("Temperature 8",TEMP_FAHRENHEIT,
              TYPE_SENSOR, DEVICE_CLASS_TEMPERATURE, "mdi:thermometer", 0),

**Comment by bsmith76s** [comment]
> Here is what I did to get this working.
> 
> This is by no means the best way to solve the issue.
> 
> In my case I wanted the battery status of my WN34 Soil Temperature.
> 
> This assumes you will never use Temperature 8 or soilbatt1.
> 
> Make a backup of file const.py before you proceed!
> 
> in the const.py file locate line 9 add a new line that says
> 
> ```
> TEMP_FAHRENHEIT,
> ```
> 
> so the import section will look like this:
> 
> ```
> TEMP_CELSIUS,
> TEMP_FAHRENHEIT,
> ```
> 
> Locate the line TYPE_TEMP8C = "temp8c" and change it to TYPE_TEMP8C = "tf_ch1"
> Locate the line TYPE_SOILBATT1 = "soilbatt1" and change it to TYPE_SOILBATT1 = "tf_batt1"
> 
> Locate the line
> 
> TYPE_TEMP8C: ("Temperature 8", TEMP_CELSIUS,
> TYPE_SENSOR, DEVICE_CLASS_TEMPERATURE, "mdi:thermometer", 0),
> 
> change it to:
> 
> ```
> TYPE_TEMP8C: ("Temperature 8", **TEMP_FAHRENHEIT**,
>               TYPE_SENSOR, DEVICE_CLASS_TEMPERATURE, "mdi:thermometer", 0),
> ```
> 
> Restart Home Assistant and the soil temperature from the WN34 will show up under sensor.temperature_8 and the battery status of the WN34 Sensor will show up as sensor.soil_moisture_1_battery.
> 
> ![Capture](https://user-images.githubusercontent.com/89822625/133528378-8f489e8e-639d-45d1-a080-620c362e53ce.JPG)

Thank you this worked for me. I only used temperature part of this I have no need for battery at the moment. Over the winter I hope to help decode the WN34 for rtl_433, I use it primarily for weather and sensor data.  I only use the Ecowitt Gateway now for testing but has been working good with Home Assistant Integration.

**Comment by lgeitner** [comment]
You're Welcome.

**Comment by Pres499** [comment]
Here is what I changed from the update from 3 days ago to get the correct soil temp.
Please backup the const.py file before you start.

From lgeitner's previous post this was added:

"in the const.py file locate line 9 add a new line that says

TEMP_FAHRENHEIT,

so the import section will look like this:

TEMP_CELSIUS,
TEMP_FAHRENHEIT,

end of lgeitner's quote

Then line 569 was changed to:
    TYPE_WN34TEMP1C: ("Soil Temperature 1", TEMP_FAHRENHEIT,
                      TYPE_SENSOR, DEVICE_CLASS_TEMPERATURE, "mdi:thermometer", 0),

Restart the server and the correct value for soil temperature should be updated.
Battery value for the soil sensor is unchanged and correct.

**Comment by bsmith76s** [comment]
> Here is what I changed from the update from 3 days ago to get the correct soil temp. Please backup the const.py file before you start.
> 
> From lgeitner's previous post this was added:
> 
> "in the const.py file locate line 9 add a new line that says
> 
> TEMP_FAHRENHEIT,
> 
> so the import section will look like this:
> 
> TEMP_CELSIUS, TEMP_FAHRENHEIT,
> 
> end of lgeitner's quote
> 
> Then line 569 was changed to: TYPE_WN34TEMP1C: ("Soil Temperature 1", TEMP_FAHRENHEIT, TYPE_SENSOR, DEVICE_CLASS_TEMPERATURE, "mdi:thermometer", 0),
> 
> Restart the server and the correct value for soil temperature should be updated. Battery value for the soil sensor is unchanged and correct.

you are correct. I feel this is what needs added/changed for this to work. I am currently on the PoppyPop fork with similar setup and it is woking correctly.

**Comment by garbled1** [comment]
Can you please try reinstalling master?  I've bumped the dependency on pyecowitt, so it should pull a new one of those down too.

**Comment by Pres499** [comment]
Yes, upgraded to .7 and Soil Temp adds a new sensor on Lovelace as "Soil Temperature 1" with the correct value.
Soil Temperature 1 Battery also displayed.
Thank you!

**Comment by garbled1** [comment]
Awesome, thanks!
